icu4c/source/i18n/datefmt.cpp

```diff
@@ -279,9 +279,8 @@ UnicodeString&
 DateFormat::format(UDate date, UnicodeString& appendTo, FieldPosition& fieldPosition) const {
     if (fCalendar != nullptr) {
         UErrorCode ec = U_ZERO_ERROR;
-        const auto* calType = fCalendar->getType();
         // Avoid a heap allocation and corresponding free for the common case
-        if (uprv_strcmp(calType, "gregorian") == 0) {
+        if (GregorianCalendar::getStaticClassID() == fCalendar->getDynamicClassID()) {
```

Comment:
optional: could we use standard [typeid](https://en.cppreference.com/w/cpp/language/typeid) instead of our old RTTI?